plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -2022,42 +2084,46 @@ def provision_device(self):
             site name hierarchy, RF profile, hostname, and AP type. Logs details
             and handles
         """
-
         provision_status = "failed"
         provision_details = None
-        try:
-            site_name_hierarchy = self.have.get("site_name_hierarchy")
-            rf_profile = self.want.get("rf_profile")
-            host_name = self.have.get("hostname")
-            type_name = self.have.get("ap_type")
-
-            if not site_name_hierarchy or not rf_profile or not host_name:
-                error_msg = ("Cannot provision device: Missing parameters - "
-                             "site_name_hierarchy: {0}, rf_profile: {1}, host_name: {2}"
-                             .format(site_name_hierarchy, rf_profile, host_name))
-                self.log(error_msg, "ERROR")
-                self.module.fail_json(msg=error_msg)
-
-            provision_params = [{
-                "rfProfile": rf_profile,
-                "deviceName": host_name,
-                "type": type_name,
-                "siteNameHierarchy": site_name_hierarchy
-            }]
-            self.log('Current device details: {0}'.format(self.pprint(provision_params)), "INFO")
+        site_name_hierarchy = self.have.get("site_name_hierarchy")
+        rf_profile = self.want.get("rf_profile")
+        host_name = self.have.get("hostname")
+        type_name = self.have.get("ap_type")
+
+        if not site_name_hierarchy or not rf_profile or not host_name:
```

Comment:
`if not all([site_name_hierarchy, rf_profile, host_name]):
`

Reply:
changed